File: frontend/app/types/analytics.ts
import type { DashboardStats } from '~/types/admin'

export type DateRangePreset =
  | 'today'
  | 'yesterday'
  | 'last_7_days'
  | 'last_30_days'
  | 'this_month'
  | 'last_month'
  | 'custom'

export type TimeSeriesPoint = {
  date: string
  orders: number
  revenue: string
}

export type CategoryBreakdown = {
  category_name: string
  order_count: number
  revenue: string
}

export type PopularItem = {
  item_name: string
  quantity: number
  revenue: string
}

export type StatusBreakdown = {
  status: string
  count: number
}

export type HourlyBreakdown = {
  hour: number // 0-23
  orders: number
  revenue: string
}

export type NoShowStats = {
  total_reservations: number
  no_shows: number
  rate: number // percent, 0-100
}

export type AnalyticsResponse = {
  preset: string
  start_date: string
  end_date: string
  summary: DashboardStats
  revenue_over_time: TimeSeriesPoint[]
  orders_over_time: TimeSeriesPoint[]
  orders_by_category: CategoryBreakdown[]
  popular_items: PopularItem[]
  order_status_distribution: StatusBreakdown[]
  sales_by_hour: HourlyBreakdown[]
  no_show: NoShowStats
}

export const DATE_RANGE_OPTIONS: { value: DateRangePreset; label: string }[] = [
  { value: 'today', label: 'Today' },
  { value: 'yesterday', label: 'Yesterday' },
  { value: 'last_7_days', label: 'Last 7 days' },
  { value: 'last_30_days', label: 'Last 30 days' },
  { value: 'this_month', label: 'This month' },
  { value: 'last_month', label: 'Last month' },
  { value: 'custom', label: 'Custom range' },
]

File: backend/app/schemas/analytics.py
"""Analytics response schemas."""

from decimal import Decimal

from pydantic import BaseModel

from app.models.enums import OrderStatus
from app.schemas.admin import DashboardStats


class TimeSeriesPoint(BaseModel):
    date: str
    orders: int = 0
    revenue: Decimal = Decimal("0.00")


class CategoryBreakdown(BaseModel):
    category_name: str
    order_count: int
    revenue: Decimal


class PopularItem(BaseModel):
    item_name: str
    quantity: int
    revenue: Decimal


class StatusBreakdown(BaseModel):
    status: OrderStatus
    count: int


class AnalyticsResponse(BaseModel):
    preset: str
    start_date: str
    end_date: str
    summary: DashboardStats
    revenue_over_time: list[TimeSeriesPoint]
    orders_over_time: list[TimeSeriesPoint]
    orders_by_category: list[CategoryBreakdown]
    popular_items: list[PopularItem]
    order_status_distribution: list[StatusBreakdown]
